### ISSUE TITLE
md.h does not include the config file, resulting in a pointless macro test

### ISSUE THREAD

**Issue by aaronmdjones** [ARMmbed/mbedtls#1001]
https://github.com/ARMmbed/mbedtls/blob/01ba45b1392a/include/mbedtls/md.h#L121-L143

Found by `clang -Weverything`: the second macro test is never true, resulting in the compiler warning that the documentation states the function is deprecated, but the function does not have the `deprecated` attribute.

**Comment by RonEld** [comment]
Hi @aaronmdjones The second macro test (`#if defined(MBEDTLS_DEPRECATED_WARNING)`) should be defined in [config.h](https://github.com/ARMmbed/mbedtls/blob/development/include/mbedtls/config.h#L180). 
From this definition documentation:
~~~
/**
 * \def MBEDTLS_DEPRECATED_WARNING
 *
 * Mark deprecated functions so that they generate a warning if used.
 * Functions deprecated in one version will usually be removed in the next
 * version. You can enable this to help you prepare the transition to a new
 * major version by making sure your code is not using these functions.
 *
 * This only works with GCC and Clang. With other compilers, you may want to
 * use MBEDTLS_DEPRECATED_REMOVED
 *
 * Uncomment to get warnings on using deprecated functions.
 */
~~~

**Comment by aaronmdjones** [comment]
@RonEld Yes, and as the title of the issue indicates, `md.h` does not include the config file.

**Comment by RonEld** [comment]
@aaronmdjones Thank you for your point which I missed.

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-1537

**Comment by gilles-peskine-arm** [comment]
@aaronmdjones  Thanks for reporting this. I believe this is fixed by #1055 which is now merged into the development branch and therefore will be included in the next release.
